### ISSUE TITLE
Any way to hide a column?

### ISSUE THREAD

**Issue by mihaibrana** [komarovalexander/ka-table#94]
Hello :). Been using your light-weight component for a few weeks now. It's been a nice experience. Good work! :). Love the hassle-free, declarative setup you created.

I'm still new to React and perhaps I'm missing some key JavaScript concepts... but... I'm having some trouble easily hiding columns.

As you indicate in the docs, I'm using an object to initialize the table. But if I want to hide a column, the only way I could think of is to pass in a different initialization object. Ideally, I'd say a column should have a visible property that can be changed with via the childComponents object.

**Comment by komarovalexander** [comment]
Hi @mihaibrana, 
Thanks for the question, I don't have plans to add columns settings to childComponent because in this object you can control real components which are rendered inside the table. And there is no column component, there are only rows and cells inside them.
but your question about the *visible* property is quite reasonable - I think you are right it would be better to have this property to avoid additional manipulations with arrays and to store hidden column settings in current properties.  Furthermore, there should be actions to hide and show columns out-of-the-box. Let me know your case I think I can add something useful in the next versions and keep this issue open until the fix (I expect the visible property can be available in 1-2 weeks).

at this time, you are right, columns array should be modified to remove a particular item from the UI. You can do it in different ways:
1) just filter columns: 
```
   <Table
      {...tableProps}
      columns={tableProps.columns.filter(c => c.key !== columnKey)} // show all columns except columnKey value
      dispatch={dispatch}
    />
```
https://stackblitz.com/edit/table-overview-ts-ybnevs?file=Demo.tsx

2)  or more 'true' ka-table way is custom reducer:
https://stackblitz.com/edit/table-overview-ts-fvjwt4?file=Demo.tsx

let me know if these approaches do not work for you

thanks for using it and providing so valuable feedback!

**Comment by mihaibrana** [comment]
Working (or playing, depending on your perception) late I see :). 

Thank you for the quick (and educative) response! 

Both these approaches are fine. I see that the reducer approach also uses .filter, but is slightly more verbose. I suppose that the reducer is better because it cuts the data before it even reaches the point when it needs to be hidden. Is this correct?

Sure, the childComponents idea was just a thought-starter, I suspected I am probably way off.

Any of these approaches will work for me.

Good job for writing a component that can be so easily tweaked :). I like your demo site too. Very useful!

**Comment by komarovalexander** [comment]
Yes, you are right. Reducer cuts colum data from tableProps. I agree that approach with visible property would be much better. It will be added in next versions. I just was waiting the request for this feature😉

**Comment by komarovalexander** [comment]
Hi @mihaibrana 

column.visible property is available starting from 5.5.0 version

demo: https://komarovalexander.github.io/ka-table/#/column-settings
release: https://github.com/komarovalexander/ka-table/releases/tag/5.5.0
